pkg/apiserver/logsearch/models.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logsearch
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"os"
+
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/dbstore"
+	"github.com/pingcap/kvproto/pkg/diagnosticspb"
+)
+
+type TaskState string
+
+const (
+	StateRunning  TaskState = "running"
+	StateCanceled TaskState = "canceled"
+	StateFinished TaskState = "finished"
+)
+
+type SearchLogRequest diagnosticspb.SearchLogRequest
+
+func (r *SearchLogRequest) Scan(src interface{}) error {
+	return json.Unmarshal([]byte(src.(string)), r)
+}
+
+func (r *SearchLogRequest) Value() (driver.Value, error) {
+	val, err := json.Marshal(r)
+	return string(val), err
+}
+
+type Component struct {
+	ServerType string `json:"server_type"`
+	IP         string `json:"ip"`
+	Port       string `json:"port"`
+	StatusPort string `json:"status_port"`
+}
+
+type TaskModel struct {
+	ID          string            `json:"task_id" gorm:"primary_key type:char(36)"`
+	TaskGroupID string            `json:"task_group_id" gorm:"type:char(36)"`
+	Component   *Component        `json:"component" gorm:"embedded"`
+	Request     *SearchLogRequest `json:"request" gorm:"type:text"`
+	State       TaskState         `json:"state"`
+	SavedPath   string            `json:"saved_path" gorm:"type:text"`
+	Error       string            `json:"error" gorm:"type:text"`
+	CreateTime  int64             `json:"create_time"`
+	StartTime   int64             `json:"start_time"`
+	StopTime    int64             `json:"stop_time"`
+}
+
+func (TaskModel) TableName() string {
+	return "log_search_tasks"
+}
+
+type TaskGroupModel struct {
+	ID      string            `json:"task_group_id" gorm:"primary_key;type:char(36)"`
+	Request *SearchLogRequest `json:"request" gorm:"type:text"`
+	State   TaskState         `json:"state"`
+}
+
+func (TaskGroupModel) TableName() string {
+	return "log_search_task_groups"
+}
+
+type PreviewModel struct {
+	ID          uint                   `json:"-" grom:"primary_key"`
+	TaskID      string                 `json:"task_id" gorm:"index;type:char(36)"`
```

Comment:
how about using simple auto increment int?

Reply:
It will increase code complexity